documentation/general/design/dash-sonic-hld.md

```diff
@@ -0,0 +1,467 @@
+```
+
+### 3.2.6 APPLIANCE
+	
+```
+DASH_APPLIANCE:{{appliance_id}} 
+    "sip": {{ip_address}}
+    "vm_vni": {{list of vnis}}
+```
+
+```
+key                      = DASH_APPLIANCE:id ; attributes specific for the appliance
+; field                  = value 
+sip                      = source ip address, to be used in encap
+vm_vni                   = list of VNIs that need special treatment. Chosen for setting direction etc. 
+```
+
+### 3.2.7 ROUTE TABLE
+
+``` 
+DASH_ROUTE_TABLE:{{eni}}:{{prefix}} 
+    "action_type": {{routing_type}} 
+    "vnet":{{vnet_name}} (OPTIONAL)
+    "appliance":{{appliance_id}} (OPTIONAL)
+    "overlay_ip":{{ip_address}} (OPTIONAL)
+    "underlay_ip":{{ip_address}} (OPTIONAL)
+    "overlay_sip":{{ip_address}} (OPTIONAL)
+    "underlay_dip":{{ip_address}} (OPTIONAL)
+    "customer_addr":{{ip_address}} (OPTIONAL)
+    "metering_bucket": {{bucket_id}}(OPTIONAL) 
+```
+  
+```
+key                      = DASH_ROUTE_TABLE:eni:prefix ; ENI route table with CA prefix
+; field                  = value 
+action_type              = routing_type              ; reference to routing type
+vnet                     = vnet name                 ; vnet name if routing_type is {vnet, vnet_direct}
+appliance                = appliance id              ; appliance id if routing_type is {appliance} 
+overlay_ip               = ip_address                ; overlay_ip to override if routing_type is {servicetunnel}, use dst ip from packet if not specified
+underlay_ip              = ip_address                ; underlay_ip to override if routing_type is {servicetunnel}, use dst ip from packet if not specified
+overlay_sip              = ip_address                ; overlay_sip if routing_type is {servicetunnel}  
+underlay_sip             = ip_address                ; overlay_sip if routing_type is {servicetunnel}
+customer_addr            = ip_address                ; CA address if routing_type is {vnet_direct}
+metering_bucket          = bucket_id                 ; metering and counter
+```
+
+### 3.2.8 VNET MAPPING TABLE
+
+``` 
+DASH_MAPPING_TABLE:{{vnet}}:{{ip_address}} 
+    "routing_type": {{routing_type}} 
+    "underlay_ip":{{ip_address}}
+    "mac_address":{{mac_address}} (OPTIONAL) 
+    "metering_bucket": {{bucket_id}}(OPTIONAL)
+```
+```
+key                      = DASH_ROUTE_TABLE:eni:ip_address ; ENI route table with CA IP
+; field                  = value 
+action_type              = routing_type              ; reference to routing type
+underlay_ip              = ip_address                ; PA address for the CA
+mac_address              = MAC address as string     ; Inner dst mac
+metering_bucket          = bucket_id                 ; metering and counter
+```
+
+## 3.3 Module Interaction
+
+A high-level module interaction is captured in the following diagram.
+
+  ![dash-high-level-diagram](https://github.com/prsunny/DASH/blob/main/Assets/dash-high-level-design.svg)
```

Comment:
Please use a relative URL for this repo, this points to a personal dev branch

Reply:
@chrispsommers Verified. @prsunny fixed all the links. 
Would you mind to approve the PR? 
Thanks